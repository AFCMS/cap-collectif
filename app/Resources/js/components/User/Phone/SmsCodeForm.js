import * as React from 'react';
import { FormattedMessage } from 'react-intl';
import { reduxForm, Field } from 'redux-form';
import { connect } from 'react-redux';
import { Button } from 'react-bootstrap';
// import UserActions from '../../../actions/UserActions';
// import FlashMessages from '../../Utils/FlashMessages';

type Props = {
  onSubmitSuccess: Function,
  submitting: boolean,
  handleSubmit?: Function,
};

// const onSubmit = (values, dispatch, props) => {
//   const { onSubmitSuccess} = this.props;
//
//
// };

const validate = ({ code }: Object) => {
  const errors = {};
  if (code === 0 || code !== 6) {
    errors.code = 'phone.confirm.constraints.code';
  }

  return errors;
};

export const formName = 'SmsCodeForm';

export class SmsCodeForm extends React.Component<Props> {
  // handleSubmit(e) {
  //   const { onSubmitSuccess } = this.props;
  //   e.preventDefault();
  //   if (this.isValid()) {
  //     const form = JSON.parse(JSON.stringify(this.state.form));
  //     this.setState({ isSubmitting: true });
  //     UserActions.sendSmsCode(form)
  //       .then(() => {
  //         this.setState(this.getInitialState());
  //         onSubmitSuccess();
  //       })
  //       .catch(error => {
  //         const response = error.response;
  //         const errors = this.state.errors;
  //         if (response.message === 'sms_code_invalid') {
  //           errors.code = ['phone.confirm.code_invalid'];
  //           this.setState({ errors, isSubmitting: false });
  //         }
  //       });
  //   }
  // }

  // renderFormErrors(field) {
  //   const errors = this.getErrorsMessages(field);
  //   if (errors.length === 0) {
  //     return null;
  //   }
  //   return <FlashMessages errors={errors} form />;
  // }

  render() {
    const { submitting, handleSubmit, onSubmitSuccess } = this.props;

    console.log(onSubmitSuccess);

    return (
      <form onSubmit={handleSubmit} style={{ maxWidth: '350px' }}>
        <Field
          type="text"
          name="code"
          autoFocus
          label={<FormattedMessage id="phone.confirm.code" />}
          id="_code"
        />
        <Button type="submit" bsStyle="primary" style={{ padding: '6px 12px 7px' }} disabled>
          {submitting ? (
            <FormattedMessage id="global.loading" />
          ) : (
            <FormattedMessage id="phone.confirm.validate" />
          )}
        </Button>
      </form>
    );
  }
}

// const mapStateToProps = (state: State, props: Props) => ({
//   initialValues: {
//     code: props..,
//   },
// });

export default connect()(
  reduxForm({
    validate,
    // onSubmit,
    form: formName,
  })(SmsCodeForm),
);
