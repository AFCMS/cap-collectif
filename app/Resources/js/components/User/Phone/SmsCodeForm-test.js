// @flow
/* eslint-env jest */
import * as React from 'react';
import { shallow } from 'enzyme';

import { SmsCodeForm } from './SmsCodeForm';

// eslint-disable-next-line react/prop-types
const props = {
  onSubmitSuccess: jest.fn(),
  submitting: false,
  handleSubmit: jest.fn(),
};

describe('<SmsCodeForm />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SmsCodeForm {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
