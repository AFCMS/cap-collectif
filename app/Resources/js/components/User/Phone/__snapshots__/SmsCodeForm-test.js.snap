// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SmsCodeForm /> should render correctly 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "maxWidth": "350px",
    }
  }
>
  <Field
    autoFocus={true}
    component={[Function]}
    id="_code"
    label={
      <FormattedMessage
        id="phone.confirm.code"
        values={Object {}}
      />
    }
    name="code"
    type="text"
  />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="primary"
    disabled={false}
    style={
      Object {
        "padding": "6px 12px 7px",
      }
    }
    type="submit"
  >
    <FormattedMessage
      id="phone.confirm.validate"
      values={Object {}}
    />
  </Button>
</form>
`;
